modules/core/src/main/java/org/apache/ignite/internal/util/io/NaiveVarInts.java

```diff
@@ -20,12 +20,17 @@
 import java.io.DataInput;
 import java.io.DataOutput;
 import java.io.IOException;
+import org.apache.ignite.internal.util.VarIntUtils;
 
 /**
- * Utils to read/write variable length ints.
+ * Utils to read/write naive variable length ints.
+ *
+ * <p>The 'naivety' relates to a simple algorithm. The naive varints produce more compact results than the 'general purpose' varints
+ * (see {@link VarIntUtils}) when the input integer is approximately between 127 and 254, so they seem to be more appropriate
+ * for small values (like lengths of short strings and collections).
  */
-public class VarInts {
-    private VarInts() {
+public class NaiveVarInts {
```

Comment:
You are probably talking about VLQ https://en.wikipedia.org/wiki/Variable-length_quantity

But you could, technically, encode integers in a variable number of bytes, in another way (that is implemented in this class). If the name (naive var ints) is confusing, let's invent another name. Do you have suggestions?

Reply:
> You are probably talking about VLQ

No, I was talking about the encoding implemented in `VarIntUtils`. As for the name for this class, I would suggest discussing it with the team, because I don't have any constructive suggestions =(

Reply:
Isn't what we have in `VarIntUtils` a variant of VLQ?

Reply:
yes, but I think that when you say "varint" to anyone else related to the project, they will think of *this particular* encoding variant. So having "naive varint" is very confusing (at least to me)

Reply:
After a bit of thinking, I realized that the advantage that `NaiveVarInts` provide is so tiny that it does not justify the complexity of having 2 implementations. I switched to usual varints and added a TODO to remove `NaiveVarInts` altogether (it will require more actions than just removing the class, so it's better to do it in a separate PR).